test/e2e/apimachinery/aggregator.go

```diff
@@ -218,7 +218,7 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
 			Name:         "sample-apiserver",
 			VolumeMounts: mounts,
 			Args: []string{
-				"--etcd-servers=http://localhost:2379",
+				"--etcd-servers=http://127.0.0.1:2379",
```

Comment:
does this break in ipv6-only environments?

Reply:
I was wondering if someone would ask this question :) I can unfortunately not answer it, does ipv6 only mean there is no 127.0.0.1 address on the lo interface in the pod network namespace anymore?

Also if we want to support ipv6 only in the conformance tests, shouldn't we then verify it?

Reply:
/cc @aojea 
this might require ipv6 detection.

Reply:
But do conformance tests pass on ipv6-only clusters (as opposed to dualstack clusters) or are there plans to make them pass on ipv6 only clusters? If the answer to both of these is no, then IMHO the answer to Jordans question doesn't matter, as this PR only changes test code.

Reply:
looks like this passed on `pull-kubernetes-conformance-kind-ipv6`... @BenTheElder, do you know if that's dual-stack or ipv6-only?

Reply:
ipv6-only, also that name is bad, it's non-serial only.

Reply:
@aojea we need to fix that, either it should be renamed or it should be made slower & include all conformance tests

Reply:
> Also if we want to support ipv6 only in the conformance tests, shouldn't we then verify it?

Yes. KIND at least is IPv6 conformant. Please don't break that. It took a while to get this corrected, and we have release blocking signal running these tests.

Reply:
https://testgrid.k8s.io/conformance-kind#kind%20(IPv6),%20master%20(dev)

(red is from a bogus test case that was reverted, was failing in all conformance jobs kind or otherwise)

Reply:
I think we're still getting ipv4 loopback available, the logs show that the pods have v6 IPs. 
I'm not sure if we should be depending on that though ...

The nodes are also dualstack-ish (they have v4 & v6 IPs, but kubernetes is only being made aware of the v6).

Pure IPv6 at the OS level sounds unlikely to me, but I'm not certain... @aojea 
At the Kubernetes level we're absolutely doing Pure IPv6 as far as Kubernetes is concerned, so anything that needs to go between pods etc. needs to be aware for sure.

Reply:
I've fixed a similar one recently in kubeadm https://github.com/kubernetes/kubernetes/pull/85745.
Unfortunately, as you can see, we don't have a pure ipv6 environment for testing and sometimes these things can escape the tests, let me try to fix the kind ipv6 job and come back to this PR

Reply:
@alvaroaleman I've added a helper method to the e2e framework for this cases
https://github.com/kubernetes/kubernetes/blob/c1b2ecaada7419d8c0efa43c37f07770e3d9583d/test/e2e/network/dual_stack.go#L216
I'd do something like:
```
etcdLocalhostAddress := "127.0.0.1"
if framework.TestContext.ClusterIsIPv6() {
			etcdLocalhostAddress = "::1"
}
etcdUrl := fmt.Sprintf("https://%s", net.JoinHostPort(etcdLocalhostAddress, "2379"))
```